rfc-123-github-actions-ci.md

```diff
@@ -0,0 +1,67 @@
+# RFC-119: Use GitHub Actions for CI
+
+## Summary
+
+This proposes using [GitHub Actions](https://github.com/features/actions)  as our preferred CI tool where applicable.
+
+CI in this case is defined as "running builds for tests and/or linting, on branches and PRs". This does not cover *deployment* of the master branch, which will continue to happen using Jenkins (until such time as we replatform (see RFC 117)) and will be considered as part of wider built pipeline work (such as continuous deployment).
+
+TechOps [started a discovery into CI tooling](https://github.com/alphagov/reliability-engineering/pull/106) a while ago, but it's not reached a conclusion yet. Now we've procured GitHub Enterprise Cloud, making use of the tools available there reduces platform complexity.
+
+## Problem
+
+We currently have many solutions for running branch builds in GOV.UK, including:
+
+- Jenkins CI (self-hosted, behind IP and GitHub team membership restrictions)
+- Travis CI (hosted, paid on a GPC card)
+- Concourse CI (self-hosted in TechOps, behind IP and GitHub team membership restrictions)
+- GitHub Actions CI (hosted, paid through GDS' GitHub Enterprise Cloud subscription)
+
+## Proposal
+
+To switch to GitHub Actions CI for continuous integration (that is, branch and PR builds for tests/linting) as the preferred approach where we do not need the wider platform integration of Jenkins.
+
+This would involve working towards GOV.UK projects only using either GitHub Actions or Jenkins for CI, no projects should use Concourse, Travis or another tool for CI without reasons to justify an exception.
+
+This would be a gradual process, because we have many repos, but this decision would deem the usage of Travis CI as a legacy approach to CI for GOV.UK.
+
+### Pros
+
+- A simpler, more modern approach compared to Jenkins.
+- GDS as a whole pays for GitHub Enterprise Cloud, which entitles us to unlimited minutes on public repos (both Linux and macOS runners), and 50,000 minutes per month of Actions on private repos.
+- Supported by GitHub. We get Enterprise support via our GitHub Enterprise Cloud subscription. And we don't have to host any of our own infrastructure.
+- Well documented, easy to [get started][actions-get-started] building a new pipeline. Easier than Concourse for CI as PR builds are built-in to GitHub rather than needing to use third-party code]().
```

Comment:
Nitpick - are these brackets meant to be here?

```suggestion
- Well documented, easy to [get started][actions-get-started] building a new pipeline. Easier than Concourse for CI as PR builds are built-in to GitHub rather than needing to use third-party code.
```

Reply:
Agh, I thought I meant to include a link but it turns out I didn't.